protocols/virtualdefund/virtualdefund.go

```diff
@@ -360,50 +360,48 @@ func (o Objective) validateSignature(sig state.Signature, participantIndex uint)
 	return true, nil
 }
 
-// Update receives an protocols.ObjectiveEvent, applies all applicable event data to the VirtualDefundObjective,
-// and returns the updated state.
-func (o Objective) Update(event protocols.ObjectiveEvent) (protocols.Objective, error) {
-	if o.Id() != event.ObjectiveId {
-		return &o, fmt.Errorf("event and objective Ids do not match: %s and %s respectively", string(event.ObjectiveId), string(o.Id()))
-	}
-
+func (o Objective) UpdateWithState(ss state.SignedState) (protocols.Objective, error) {
```

Comment:
Somewhat outside of scope:

I think we could make all objective methods have a pointer receiver.